src/messages/embed.ts

```diff
@@ -0,0 +1,260 @@
+import type {
+	APIEmbed,
+	APIEmbedAuthor,
+	APIEmbedField,
+	APIEmbedFooter,
+	APIEmbedImage,
+	APIEmbedProvider,
+	APIEmbedThumbnail,
+	APIEmbedVideo,
+} from 'discord-api-types/v8';
+
+/**
+ * Represents an embed in a message (image/video preview, rich embed, etc.)
+ */
+export class Embed implements APIEmbed {
+	/**
+	 * An array of Fields of this Embed.
+	 */
+	public fields: APIEmbedField[] = [];
+
+	/**
+	 * The embed title.
+	 */
+	public title?: string;
+
+	/**
+	 * The embed description.
+	 */
+	public description?: string;
+
+	/**
+	 * The embed url.
+	 */
+	public url?: string;
+
+	/**
+	 * The embed color.
+	 */
+	public color?: number;
+
+	/**
+	 * The timestamp of the embed in the ISO format.
+	 */
+	public timestamp?: string;
+
+	/**
+	 * The embed thumbnail data.
+	 */
+	public thumbnail?: APIEmbedThumbnail;
+
+	/**
+	 * The embed image data.
+	 * @since 0.0.1
+	 */
+	public image?: APIEmbedImage;
+
+	/**
+	 * Received video data.
+	 * @since 0.0.1
+	 */
+	public video?: APIEmbedVideo;
+
+	/**
+	 * The embed author data.
+	 * @since 0.0.1
+	 */
+	public author?: APIEmbedAuthor;
+
+	/**
+	 * Received data about the embed provider.
+	 * @since 0.0.1
+	 */
+	public provider?: APIEmbedProvider;
+
+	/**
+	 * The embed footer data.
+	 */
+	public footer?: APIEmbedFooter;
+
+	public constructor(data: APIEmbed = {}) {
+		this.title = data.title;
+		this.description = data.description;
+		this.url = data.url;
+		this.color = data.color;
+		this.thumbnail = data.thumbnail;
+		this.image = data.image;
+		this.video = data.video;
+		this.author = data.author;
+		this.provider = data.provider;
+		this.footer = data.footer;
+
+		if (data.timestamp) this.timestamp = new Date(data.timestamp).toISOString();
+		if (data.fields) this.fields = data.fields;
```

Comment:
And maybe move it alongside the other fields.
```suggestion
		this.fields = data.fields ?? [];
```